generators/docker-compose/prompts.js

```diff
@@ -167,8 +168,69 @@ function askForElk() {
     }.bind(this));
 }
 
+function askForServiceDiscovery() {
+    if (this.regenerate) return;
+
+    var done = this.async();
+
+    var serviceDiscoveryEnabledApps = [];
+    this.appConfigs.forEach(function (appConfig, index) {
+        if(appConfig.serviceDiscoveryType !== 'no' && (appConfig.applicationType === 'microservice' || appConfig.applicationType === 'gateway')) {
+            serviceDiscoveryEnabledApps.push({baseName: appConfig.baseName, serviceDiscoveryType: appConfig.serviceDiscoveryType});
+        }
+    }, this);
+
+    if(serviceDiscoveryEnabledApps.length === 0) {
+        return;
+    }
+
+    if(serviceDiscoveryEnabledApps.every(app => app.serviceDiscoveryType === 'consul')){
+        this.serviceDiscoveryType = 'consul';
+        this.log(chalk.green('Consul detected as the service discovery and configuration provider used by your apps'));
+        done();
+    }
+    else if(serviceDiscoveryEnabledApps.every(app => app.serviceDiscoveryType === 'eureka')){
+        this.serviceDiscoveryType = 'eureka';
+        this.log(chalk.green('JHipster registry detected as the service discovery and configuration provider used by your apps'));
+        done();
+    }
+    else {
+        this.log(chalk.yellow('Unable to determine the service discovery and configuration provider to use from your apps configuration.'));
+        this.log('Your service discovery enabled apps:');
+        serviceDiscoveryEnabledApps.forEach(function(app){
+            this.log('  - ' + app.baseName + ' ('  + app.serviceDiscoveryType + ')');
```

Comment:
you could do `` `  -${app.baseName} (${app.serviceDiscoveryType})` `` since we are on es6